### DISCUSSION TITLE
Regular snapshots of the full orderbook in cryptofeed 2.0

### DISCUSSION THREAD

**Discussion by agijsberts** [bmoscon/cryptofeed#648] [Q&A] [answered]
Feeds in `cryptofeed` 1.9.x accepted a `book_interval` argument that would ensure that a full book snapshot was sent after a certain number of delta updates. These snapshots were useful in my usecase, as it allows to start replaying the order book stream from any intermediate point in time (at a granularity of `book_interval` of course).

Commit https://github.com/bmoscon/cryptofeed/commit/592cffe27b7151b76df0488fbab4b5da3c0e30cd removed this functionality as part of the order book reorganization. Am I correct to assume that we now need to implement this feature in our backends?

**Comment by bmoscon** [comment, marked as answer]
@agijsberts I added the snapshot_interval kwarg to the book backends

**Reply by agijsberts** [reply]
Thanks for your quick response, I implemented a similar strategy in my custom backend.

Just for completeness, but unless I'm mistaking your current implementation deviates slightly from the old behavior. Cryptofeed 1.9 would send the delta update *followed by* the snapshot of the full book (with identical timestamps), whereas the present implementation sends a full orderbook *instead of* a delta update. So to track the orderflow the consumer would have to recalculate the delta manually.

**Reply by bmoscon** [reply]
@agijsberts - ok I changed it so it mirrors the old behavior. Deltas then snapshot when book interval is hit

**Reply by agijsberts** [reply]
Wonderful, thank you!

**Comment by k-d-l** [comment]
Could you please give an example of using _BackendBookCallback_ with _snapshot_interval_ parameter

**Reply by bmoscon** [reply]
one exists: https://github.com/bmoscon/cryptofeed/blob/master/examples/demo_zmq.py#L40

That is for the ZMQ bookcallback.

There isn't one for BackendBookCallback because that is an interface. You'd never use that directly, you'd define a child class of it that defined its own `write` method.

**Reply by k-d-l** [reply]
How can I use it with Arctic backend?
I don't see there book callback

**Reply by bmoscon** [reply]
you can't - orderbooks arent supported on arctic

**Reply by k-d-l** [reply]
Ok, thanks
